docs/sql/statements/create-table.rst

```diff
@@ -130,6 +130,40 @@ statement that doesn't contain an explicit value for it.
 The default clause :ref:`expression <gloss-expression>` is variable-free, it
 means that subqueries and cross-references to other columns are not allowed.
 
+.. NOTE::
+
+    Default values are not allowed for columns of type ``OBJECT``, e.g.::
+
+      CREATE TABLE tbl(obj OBJECT DEFAULT {key='foo'}
+
+    but they are allowed for sub columns of an object column, e.g.::
+
+      CREATE TABLE tbl(obj OBJECT AS(key TEXT DEFAULT 'foo'))
+
+    The effect of the later, is different though, as if no value is provided for
+    the column ``obj`` then the value inserted is ``NULL``, but if a value for
+    the ``obj`` column is provided, but there is no value for ``key``
+    sub-column, then the ``key`` acquires the default value defined. e.g.::
+
+      CREATE TABLE tbl(i INT, obj OBJECT AS(key TEXT DEFAULT 'foo'))
+      INSERT INTO tbl(i) VALUES(1)
+      SELECT * FROM tbl
+      +---+------+
+      | i | obj  |
+      +---+------+
+      | 1 | NULL |
+      +---+------+
```

Comment:
On first example: I think having null is not what user would expect (I think expected is {"key": "foo"} when value is not provided) so probably saying 

> they are allowed for sub columns of an object column

is not really true. I feel like it's a bug and fixing it kinda reduces priority of https://github.com/crate/crate/issues/14189 (it will become slightly more convenient syntax sugar then)

Reply:
imho it's different semantics:
- default value on the root object means what value you set when no value at all is provided to the root object.
- default value on a sub columns means what value you set when a value for the root object is provided but no value for the sub column.

Reply:
Makes sense. 

And if users don't want to have NULL like in your first example then they will have to use default root (which should be covered by https://github.com/crate/crate/issues/14189)